sources/PPU/PPU.cpp

```diff
@@ -308,20 +301,35 @@ namespace ComSquare::PPU
 		(void)cycles;
 
 		this->renderMainAndSubScreen();
-		this->add_buffer(this->_screen, this->_subScreen);
-		this->add_buffer(this->_screen, this->_mainScreen);
-		//this->_backgrounds[2].renderBackground();
-		//add_buffer(this->_screen, this->_backgrounds[2].buffer);
+		Utils::addBuffer(this->_screen, this->_subScreen);
+		Utils::addBuffer(this->_screen, this->_mainScreen);
+
+		int i = 0;
+		int j = 0;
+		std::for_each(this->_screen.begin(), this->_screen.end(), [this, &i, &j](const auto &row) {
+			std::for_each(row.begin(), row.end(), [this, &i, &j](const auto &pixel) {
+				this->_renderer.putPixel(i, j++, pixel);
```

Comment:
Don't you think we should remove this putPixel and use something like a std::span to write multiples bytes as once with a memcpy like method?

Reply:
This would also remove the need of the weird i & j variables.

Reply:
that's a good idea but it will not be done in this pr